src/lib/dates.ts

```diff
@@ -138,3 +138,20 @@ export const createIsoDate = (date: string, timezone: string): string => {
  */
 export const getTimezoneFromIsoDate = (date: string): string | null =>
   DateTime.fromISO(date).zoneName
+
+/**
+ * Takes an ISO datetime string (2023-07-20T00:00:00.000-07:00) and returns a string of the format ${date} at ${time}
+ * (Jul 20 at 12:00 AM).
+ *
+ * @param date
+ * @returns string
+ *
+ */
+export const formatDateAndTime = (date: string): string =>
+  `${DateTime.fromISO(date).toLocaleString({
+    month: 'short',
+    day: 'numeric',
+  })} at ${DateTime.fromISO(date).toLocaleString({
+    hour: 'numeric',
+    minute: '2-digit',
+  })}`
```

Comment:
Let's use proper locale formatting and format the full date and time together, e.g., not mix locale w/ English text ("at").

Does something like this work?
```
export const formatDateAndTime = (date: string): string =>
  DateTime.fromISO(date).toLocaleString(
    month: 'short',
    day: 'numeric',
    hour: 'numeric',
    minute: '2-digit',
  )}
```

Or maybe just 

`DateTime.fromISO(date).toLocaleString(DateTime.DATETIME_FULL)`

Reply:
The figma uses`${date} at ${time}` format. I guess that part isn't critical now that I'm writing this.

Reply:
<img width="229" alt="Screenshot 2023-08-29 at 12 34 42 PM" src="https://github.com/seamapi/react/assets/11382084/2c990065-cdea-4ff3-a7ed-490204a96378">

This is probably fine

Reply:
https://github.com/seamapi/react/pull/378/commits/dd76de64d418e71e13b2152e1c6e37b43a7c91c5#diff-be9722f143d0d492db199700788ead0bc38d9b1b4bb9401e82f0f1434d8da1ceR150-R157